google/cloud/storage/bucket_metadata.cc

```diff
@@ -384,6 +384,271 @@ std::ostream& operator<<(std::ostream& os, BucketMetadata const& rhs) {
+    std::vector<BucketAccessControl> const& v) {
+  if (v.empty()) {
+    return ResetAcl();
+  }
+  std::vector<internal::nl::json> array;
+  array.reserve(v.size());
+  for (auto const& a : v) {
+    array.emplace_back(internal::nl::json{
+        {"entity", a.entity()},
+        {"role", a.role()},
+    });
+  }
+  impl_.SetArrayField("acl", array);
+  return *this;
+}
+
+BucketMetadataPatchBuilder& BucketMetadataPatchBuilder::ResetAcl() {
+  impl_.RemoveField("acl");
+  return *this;
+}
+
+BucketMetadataPatchBuilder& BucketMetadataPatchBuilder::SetBilling(
+    BucketBilling const& v) {
+  impl_.AddSubPatch("billing", internal::PatchBuilder().SetBoolField(
+                                   "requesterPays", v.requester_pays));
+  return *this;
+}
+
+BucketMetadataPatchBuilder& BucketMetadataPatchBuilder::ResetBilling() {
+  impl_.RemoveField("billing");
+  return *this;
+}
+
+BucketMetadataPatchBuilder& BucketMetadataPatchBuilder::SetCors(
+    std::vector<CorsEntry> const& v) {
+  if (v.empty()) {
+    return ResetCors();
+  }
+  std::vector<internal::nl::json> array;
+  array.reserve(v.size());
+  for (auto const& a : v) {
+    internal::nl::json entry;
+    if (a.max_age_seconds.has_value()) {
+      entry["maxAgeSeconds"] = *a.max_age_seconds;
+    }
+    if (not a.method.empty()) {
+      entry["method"] = a.method;
+    }
+    if (not a.origin.empty()) {
+      entry["origin"] = a.origin;
+    }
+    if (not a.response_header.empty()) {
+      entry["responseHeader"] = a.response_header;
+    }
+    array.emplace_back(std::move(entry));
+  }
+  impl_.SetArrayField("cors", array);
+  return *this;
+}
+
+BucketMetadataPatchBuilder& BucketMetadataPatchBuilder::ResetCors() {
+  impl_.RemoveField("cors");
+  return *this;
+}
+
+BucketMetadataPatchBuilder& BucketMetadataPatchBuilder::SetDefaultAcl(
+    std::vector<ObjectAccessControl> const& v) {
+  if (v.empty()) {
+    return ResetDefaultAcl();
+  }
+  std::vector<internal::nl::json> array;
+  array.reserve(v.size());
+  for (auto const& a : v) {
+    array.emplace_back(internal::nl::json{
+        {"entity", a.entity()},
+        {"role", a.role()},
+    });
+  }
+  impl_.SetArrayField("defaultObjectAcl", array);
+  return *this;
+}
+
+BucketMetadataPatchBuilder& BucketMetadataPatchBuilder::ResetDefaultAcl() {
+  impl_.RemoveField("defaultObjectAcl");
+  return *this;
+}
+
+BucketMetadataPatchBuilder& BucketMetadataPatchBuilder::SetEncryption(
+    BucketEncryption const& v) {
+  impl_.AddSubPatch("encryption",
+                    internal::PatchBuilder().SetStringField(
+                        "defaultKmsKeyName", v.default_kms_key_name));
+  return *this;
+}
+
+BucketMetadataPatchBuilder& BucketMetadataPatchBuilder::ResetEncryption() {
+  impl_.RemoveField("encryption");
+  return *this;
+}
+
+BucketMetadataPatchBuilder& BucketMetadataPatchBuilder::SetLabel(
+    std::string const& label, std::string const& value) {
+  if (value.empty()) {
```

Comment:
This `if` statement should be removed.  An empty string is a valid label value:

```
$ gsutil label ch -l "key":"" gs://BUCKET
Setting label configuration on gs://BUCKET/...

$ gsutil ls -Lb gs://BUCKET
gs://BUCKET/ :
	  ...
	  Labels:				
	  {
	    "key": ""
	  }
	  ...
```

Reply:
Done.